src/site/topic22.rst

```diff
@@ -263,6 +263,106 @@ Binary Tree Node
 Linked Binary Tree
 ^^^^^^^^^^^^^^^^^^
 
+* Although there will be no implementation of a general ``BinaryTree``, we can discuss what some specific tree based algorithms
+
+**Size**
+
+* If we have some arbitrary binary tree and do not know it's size (and it has no size field), how would we count the number of elements?
+
+.. code-block:: java
+    :linenos:
+
+    public int size() {
+        return size(root);
+    }
+
+    private int size(Node<T> current) {
+        if (current == null) {
+            return 0;
+        } else {
+            return 1 + size((current.getLeft())) + size(current.getRight());
+        }
+    }
+
+* The idea here is, if the current node exists, the the size of the (sub)tree will be 1 + the size of the left subtree + the size of the right subtree
+* Here we also make use of a public helper method that always starts the recursive method at the root
+
+* What is the computational complexity of ``size()``?
+    * :math:`O(n)`, where :math:`n` is the number of nodes in the tree
+
+**Contains**
+
+* If we have an arbitrary binary tree and we want to search it for a specific element
+
+.. code-block:: java
+    :linenos:
+
+    public boolean contains(T element) {
+        return contains(root, element);
+    }
+
+    private boolean contains(Node<T> current, T element) {
+        if (current == null) {
+            return false;
+        } else if (current.getData().equals(element)) {
+            return true;
+        } else {
+            boolean inSubtree = contains(current.getLeft(), element);
+            if(!inSubtree) {
+                inSubtree = contains(current.getRight(), element);
+            }
+            return inSubtree;
```

Comment:
I don't like the mutable value here, pretty weird implementation to read. More simply, `return contains(current.getLeft(), needle) || contains(current.getRight(), needle);`